src/db/RUEatsRepository.js

```diff
@@ -170,6 +170,212 @@ module.exports = class RUEatsRepository {
+                resolve(results.affectedRows > 0);
+            }
+        });
+    });
+  }
+
+  toggleItemAvailability(restaurantID, itemID, isAvailable) {
+    return new Promise((resolve, reject) => {
+        const query = `
+            UPDATE menu
+            SET is_available = ?
+            WHERE item_id = ? AND restaurant_id = ?
+        `;
+
+        this.connection.query(query, [
+            isAvailable,
+            itemID,
+            restaurantID
+        ], function (error, results, fields) {
+            if (error) {
+                reject(error);
+            } else {
+                resolve(results.affectedRows > 0);
+            }
+        });
+    });
+  }
+
+  addReview(review_title, description, stars, media, author_id, restaurant_id) {
+    return new Promise((resolve, reject) => {
+        const date_created = new Date().toISOString().slice(0, 19).replace('T', ' ');
+        const query = 'INSERT INTO reviews (review_title, description, stars, media, author_id, restaurant_id, date_created) VALUES (?, ?, ?, ?, ?, ?, ?)';
+        this.connection.query(query, [review_title, description, stars, media, author_id, restaurant_id, date_created], function (error, results) {
+            if (error) {
+                reject(error);
+            } else {
+                resolve(results.insertId);
+            }
+        });
+    });
+  }
+
+  hasUserOrderedFromRestaurant(author_id, restaurant_id) {
+    return new Promise((resolve, reject) => {
+        const query = 'SELECT EXISTS(SELECT 1 FROM orders WHERE user_id = ? AND restaurant_id = ? LIMIT 1) as hasOrdered';
+        this.connection.query(query, [author_id, restaurant_id], function (error, results) {
+            if (error) {
+                reject(error);
+            } else {
+                resolve(results[0].hasOrdered);
+            }
+        });
+    });
+  }
+
+
+  deleteMenuItem(restaurantID, itemID) {
+    return new Promise((resolve, reject) => {
+        this.connection.query('DELETE FROM menu WHERE item_id = ? AND restaurant_id = ?', [itemID, restaurantID], function (error, results) {
+            if (error) {
+                reject(error);
+            } else {
+                resolve(results);
+            }
+        });
+    });
+  }
+
+  getInsightsByRestaurantID(restaurantID) {
+    return new Promise(async (resolve, reject) => {
+        try {
+            const orderCount = await this._getOrderCount(restaurantID);
+            const averageRating = await this._getAverageRating(restaurantID);
+            const totalEarnings = await this._getTotalEarnings(restaurantID);
+            const orderStatusCounts = await this._getOrderStatusCounts(restaurantID);
+
+            resolve({
+                orderCount,
+                averageRating,
+                totalEarnings,
+                orderStatusCounts
+            });
+        } catch (error) {
+            reject(error);
+        }
+    });
+}
+
+  _getOrderCount(restaurantID) {
```

Comment:
why are you starting function name with '_' ?? is it required??

Reply:
to identify the helper functions